lib/posix/fs.c

```diff
@@ -290,8 +290,11 @@ struct dirent *readdir(DIR *dirp)
 	}
 
 	rc = strlen(fdirent.name);
-	memcpy(pdirent.d_name, fdirent.name,
-		rc <= PATH_MAX ? rc : PATH_MAX);
+	rc = (rc <= PATH_MAX) ? rc : PATH_MAX;
```

Comment:
What happens when the filesystem allows for filenames of, say, 255 characters, and PATH_MAX is defined to, say, 32 characters? What will happen when the filename in the dirent is truncated?

Reply:
yeah, but that is not the ideal case. The expectation is path name is > file name.

Reply:
If there's that expectation, please add a build time assertion.  Otherwise, this will silently fail, and anyone trying to list a directory to, say, open files, will be scratching their heads when they can't open a file that readdir() told them that it exists.